### PR TITLE
Add support for multiple map categories.

### PR DESCRIPTION
**Pull request by pchote**
This is the final backwards-incompatible change that I'd like to propose for the next release.  I've split it into its own PR because it depends on resource center changes that may not happen immediately (it needs to be able to parse and expose the categories via the map API).

This replaces the map `Type` field with a list of `Categories`, where the first category is treated as the main map type, and remaining categories are used for filtering the map chooser.  As a test case I have updated and added categories for the TS maps; this will currently break the other mods (updating those now would just end up with rebase conflicts while we wait for the resource center changes).

Depends on #10839. Only the last four commits are new.

### CODE REVIEW

**Comment by GraionDilach** [comment]
Map importers won't spit out format 11 maps.

**Comment by obrakmann** [comment]
```
OpenRA.Game/Map/MapPreview.cs:L247: [SA1026] There should be no space between the new keyword and the opening square bracket in an implicitly typed array allocation.

OpenRA.Game/Map/MapPreview.cs:L247: [SA1010] Invalid spacing around the opening square bracket.
```

**Comment by pchote** [comment]
Adding the dependencies tag back, since this requires changes to the resource center API.

**Comment by ihptru** [comment]
Resource Center implemented support at https://github.com/ihptru/OpenRA-Resources/commit/d52495495e0a1675044a83f948d586677270fe78

It's in production now.

**Comment by pchote** [comment]
Updated.

File: OpenRA.Mods.Common/Widgets/Logic/ReplayBrowserLogic.cs [hunk 1]
```diff
@@ -83,7 +83,10 @@ public ReplayBrowserLogic(Widget widget, ModData modData, Action onExit, Action
 
 			var type = panel.GetOrNull<LabelWidget>("MAP_TYPE");
 			if (type != null)
-				type.GetText = () => selectedReplay.GameInfo.MapPreview.Type;
+			{
+				var mapType = selectedReplay.GameInfo.MapPreview.Categories.FirstOrDefault() ?? "";
```
Comment (obrakmann) [inline OpenRA.Mods.Common/Widgets/Logic/ReplayBrowserLogic.cs]: This crashes with an NRE.

**Comment by pchote** [comment]
Fixed and rebased.

**Comment by obrakmann** [comment]
Looks fine, :+1: ,but already needs another rebase.

**Comment by pchote** [comment]
Rebased and changed LobbyMapPreviewLogic a bit.

**Comment by pchote** [comment]
This pr is blocking fixes for #10718 and #9684.  We need to merge it before we can make more progress towards a playtest...

**Comment by penev92** [comment]
:+1:

**Comment by GraionDilach** [comment]
eh... eh.

I ended up being ignored here. The map importers were untouched.

**Comment by pchote** [comment]
The map importers don't need to be touched...?

**Comment by GraionDilach** [comment]
Oh, those are set in the map constructors and not in the actual map importers, sorry... dammit, I shouldn't review/catch-up stuff when I'm sleepy.

**Comment by penev92** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-%28bleed%29/_compare/c1fb7ea215fbc91ea4afaeed16a9e982cb60e8f4)

**Pull request merged**